src/Model/Linter.php

```diff
@@ -1,19 +1,40 @@
 <?php
 
-namespace DTL\BehatLint\Model;
+namespace DTL\GherkinLint\Model;
 
 use Cucumber\Gherkin\GherkinParser;
+use Generator;
 
 class Linter
 {
-    public function __construct(private GherkinParser $parser)
-    {
+    public function __construct(
+        private GherkinParser $parser,
+        /**
+         * @var Rule[]
+         */
+        private array $rules
+    ) {
     }
 
-    public function lint(string $uri, string $contents): FeatureDiagnostics
+    /**
+     * @return Generator<FeatureDiagnostic>
+     */
+    public function lint(string $uri, string $contents): Generator
     {
-        $node = $this->parser->parseString($uri, $contents);
+        $envelopes = iterator_to_array($this->parser->parseString($uri, $contents));
+
+        foreach ($envelopes as $envelope) {
+            foreach ($this->rules as $rule) {
+                $document = $envelope->gherkinDocument;
+                if (null === $document) {
+                    continue;
+                }
+
+                yield from $rule->analyse($document);
+            }
 
-        return new FeatureDiagnostics([]);
+            // why multiple envelopes?
+            break;
```

Comment:
remove this and possibly ensure that we only get the "gherkin document" envelope